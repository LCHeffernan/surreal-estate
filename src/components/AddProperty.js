/* eslint-disable no-console */
import React, { useState } from "react";
// import axios from "axios";
import postFields from "../requests/postFields";
import "../styles/add-property.css";

const AddProperty = () => {
  const initialState = {
    fields: {
      title: "",
      city: "Manchester",
      type: "Detached",
      bedrooms: 1,
      bathrooms: 1,
      price: 0,
      email: "",
    },
  };
  const [fields, setFields] = useState(initialState.fields);

  const handleAddProperty = (event) => {
    postFields(fields);
    event.preventDefault();
  };

  const handleFieldChange = (event) => {
    setFields({ ...fields, [event.target.name]: event.target.value });
  };

  return (
    <div className="add-property">
      Add Property Page
      <form onSubmit={handleAddProperty}>
        <div className="form-field">
          <label htmlFor="title">
            Title
            <input
              id="title"
              name="title"
              placeholder="title"
              type="text"
              value={fields.title}
              onChange={handleFieldChange}
            />
          </label>
        </div>
        <div className="form-field">
          <label htmlFor="city">
            City
            <select
              id="city"
              name="city"
              value={fields.city}
              onChange={handleFieldChange}
            >
              <option value="Manchester">Manchester</option>
              <option value="Leeds">Leeds</option>
              <option value="Sheffield">Sheffield</option>
              <option value="Liverpool">Liverpool</option>
            </select>
          </label>
        </div>
        <div className="form-field">
          <label htmlFor="type">
            House type
            <select
              id="type"
              name="type"
              value={fields.type}
              onChange={handleFieldChange}
            >
              <option value="Flat">Flat</option>
              <option value="Detached">Detached</option>
              <option value="Semi-Detached">Semi-Detached</option>
              <option value="Terraced">Terraced</option>
              <option value="End of Terrace">End of Terrace</option>
              <option value="Cottage">Cottage</option>
              <option value="Bungalow">Bungalow</option>
            </select>
          </label>
        </div>
        <div className="form-field">
          <label htmlFor="bedrooms">
            Bedrooms
            <input
              id="bedrooms"
              name="bedrooms"
              placeholder="Bedrooms"
              type="number"
              value={fields.bedrooms}
              onChange={handleFieldChange}
            />
          </label>
        </div>
        <div className="form-field">
          <label htmlFor="bathrooms">
            Bathrooms
            <input
              id="bathrooms"
              name="bathrooms"
              placeholder="Bathrooms"
              type="number"
              value={fields.bathrooms}
              onChange={handleFieldChange}
            />
          </label>
        </div>
        <div className="form-field">
          <label htmlFor="price">
            Price
            <input
              id="price"
              name="price"
              placeholder="100000"
              type="number"
              value={fields.price}
              onChange={handleFieldChange}
            />
          </label>
        </div>
        <div className="form-field">
          <label htmlFor="email">
            email
            <input
              id="email"
              name="email"
              placeholder="example@outlook.com"
              type="text"
              value={fields.email}
              onChange={handleFieldChange}
            />
          </label>
        </div>
        <div className="form-field">
          <button type="submit">Add</button>
        </div>
      </form>
    </div>
  );
};

export default AddProperty;
