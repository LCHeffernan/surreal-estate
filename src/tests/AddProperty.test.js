import React from "react";
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import AddProperty from "../components/AddProperty";

describe("Properties", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <AddProperty />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct text", () => {
    render(
      <BrowserRouter>
        <AddProperty />
      </BrowserRouter>
    );
    const text = screen.getByText(/add property page/i);
    expect(text).toBeInTheDocument();
  });
});
