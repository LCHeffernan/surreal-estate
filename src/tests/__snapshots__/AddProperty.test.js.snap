// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Properties renders correctly 1`] = `
<DocumentFragment>
  <div
    class="add-property"
  >
    Add Property Page
    <form>
      <div
        class="form-field"
      >
        <label
          for="title"
        >
          Title
          <input
            id="title"
            name="title"
            placeholder="title"
            type="text"
            value=""
          />
        </label>
      </div>
      <div
        class="form-field"
      >
        <label
          for="city"
        >
          City
          <select
            id="city"
            name="city"
          >
            <option
              value="Manchester"
            >
              Manchester
            </option>
            <option
              value="Leeds"
            >
              Leeds
            </option>
            <option
              value="Sheffield"
            >
              Sheffield
            </option>
            <option
              value="Liverpool"
            >
              Liverpool
            </option>
          </select>
        </label>
      </div>
      <div
        class="form-field"
      >
        <label
          for="type"
        >
          House type
          <select
            id="type"
            name="type"
          >
            <option
              value="Flat"
            >
              Flat
            </option>
            <option
              value="Detached"
            >
              Detached
            </option>
            <option
              value="Semi-Detached"
            >
              Semi-Detached
            </option>
            <option
              value="Terraced"
            >
              Terraced
            </option>
            <option
              value="End of Terrace"
            >
              End of Terrace
            </option>
            <option
              value="Cottage"
            >
              Cottage
            </option>
            <option
              value="Bungalow"
            >
              Bungalow
            </option>
          </select>
        </label>
      </div>
      <div
        class="form-field"
      >
        <label
          for="bedrooms"
        >
          Bedrooms
          <input
            id="bedrooms"
            name="bedrooms"
            placeholder="Bedrooms"
            type="number"
            value="1"
          />
        </label>
      </div>
      <div
        class="form-field"
      >
        <label
          for="bathrooms"
        >
          Bathrooms
          <input
            id="bathrooms"
            name="bathrooms"
            placeholder="Bathrooms"
            type="number"
            value="1"
          />
        </label>
      </div>
      <div
        class="form-field"
      >
        <label
          for="price"
        >
          Price
          <input
            id="price"
            name="price"
            placeholder="100000"
            type="number"
            value="0"
          />
        </label>
      </div>
      <div
        class="form-field"
      >
        <label
          for="email"
        >
          email
          <input
            id="email"
            name="email"
            placeholder="example@outlook.com"
            type="text"
            value=""
          />
        </label>
      </div>
      <div
        class="form-field"
      >
        <button
          type="submit"
        >
          Add
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
